app/src/main/java/ru/vssemikoz/newsfeed/data/mappers/NewsMapper.java

```diff
@@ -0,0 +1,11 @@
+package ru.vssemikoz.newsfeed.data.mappers;
+
+import java.util.List;
+
+import retrofit2.Response;
+import ru.vssemikoz.newsfeed.models.NewsApiResponse;
+import ru.vssemikoz.newsfeed.models.NewsItem;
+
+public interface NewsMapper<P> {
```

Comment:
Why do you consider generics for this interface? This interface is not intended to be used across different mappers. It's the only usage is in the `ApiResponseMapper` and no parameters are needed for this interface.